syntax = "proto3";

service Math {
    rpc Sqrt (SqrtRequest) returns (SqrtResponse) {}
    rpc Stat (stream StatRequest) returns (StatResponse) {}
    rpc Factor (FactorRequest) returns (stream FactorResponse) {}
}

message SqrtRequest {
    double value = 1;
}

message SqrtResponse {
    double value = 1;
}

message StatRequest {
    int32 value = 1;
}

message StatResponse {
    int32 sum = 1;
    int32 count = 2;
}

message FactorRequest {
    int32 value = 1;
}

message FactorResponse {
    int32 value = 1;
}


service Math2 {
    rpc Add (AddRequest) returns (AddResponse) {}
    rpc Sub (SubRequest) returns (SubResponse) {}
}

message AddRequest {
    double a = 1;
    double b = 2;
}

message AddResponse {
    double c = 3;
}


message SubRequest {
    double a = 1;
    double b = 2;
}

message SubResponse {
    double c = 1;
}